lib/puppet/type.rb

```diff
@@ -1295,7 +1295,7 @@ def pathbuilder
       if Puppet.settings[:strict] != :off
         # Only warn if `audit` metaparam came from a manifest
         if file && line
-          Puppet.warn(_("The `audit` metaparameter is ignored."), { :line => line, :file => file })
+          Puppet.warn_once(_("The `audit` metaparameter is ignored."), { :line => line, :file => file })
```

Comment:
`warn_once` is a little tricky, you have to tell it what kind of warning as the first parameter, and `key` as the second, which serves as the message identity (so we don't emit the same warning twice).

I don't think we have a category for this kind of thing (audit is ignored as opposed to deprecated), so perhaps something like:

```ruby
Puppet.warn_once(:audit, "ignore_metaparameter", _("The `audit` metaparameter..."), ...)
```